### ISSUE TITLE
The `string` function is not working currently unless passed a literal format

### ISSUE THREAD

**Issue by DiasFranciscoA** [YoYoGames/GameMaker-Bugs#6017]
### Description

The `string(format, val1, val2, val3)`... is not working correctly when the format is not a literal.
See the example below:

```
var _format = "{0} {1} {2}";
var _result = string(_format, 12, "hello", [1, 2, 3]);
show_debug_message(_result);

var _result = string("{0} {1} {2}", 12, "hello", [1, 2, 3]);
show_debug_message(_result);
```

The first one doesn't work while the second one does work.

> [!IMPORTANT]
> This was working on the latest **April24** release. Seems to happen at the AssetCompiler level.

### Expected Change

Both approaches should work correctly

### Steps To Reproduce

1. Start GameMaker
2. Open attached project
4. See the issue

[TestProject.zip](https://github.com/YoYoGames/GameMaker-Bugs/files/15473079/TestProject.zip)

### How reliably can you recreate this issue using your steps above?

Always

### Which version of GameMaker are you reporting this issue for?

2024.600 (Betas)

### Which platform(s) are you seeing the problem on?

Windows

### Contact Us Package Attached?

- [X] I have attached my Contact Us Package

### Sample Project Added?

- [X] I have included a small sample project

**Comment by rwkay** [comment]
Fixed in 2024.6

**Comment by mgeddesGM** [comment]
verified as of  IDE v2024.600.0.569  Runtime v2024.600.0.596
